src/libraries/System.Security.Cryptography.Cose/tests/CoseMessageTests.Sign.CustomHeaderMaps.cs

```diff
@@ -132,7 +132,7 @@ public void SignWith_EmptyProtectedHeaderMap_UnprotectedHeaderMapWithAlgorithm()
             CoseHeaderMap protectedHeaders = GetEmptyHeaderMap();
             CoseHeaderMap unprotectedHeaders = GetHeaderMapWithAlgorithm(DefaultAlgorithm);
 
-            Assert.Throws<CryptographicException>(() => Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders, unprotectedHeaders));
+            Assert.Throws<ArgumentException>(() => Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders, unprotectedHeaders));
```

Comment:
Any ArgumentException tests should verify that the ParamName was correct.

```suggestion
            Assert.Throws<ArgumentException>(
                "unprotectedHeaders",
                () => Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders, unprotectedHeaders));
```

In the rare case that we actually can't determine which of two arguments was bad:

```suggestion
            ArgumentException ex = Assert.Throws<ArgumentException>(() => Sign(s_sampleContent, DefaultKey, DefaultHash, protectedHeaders, unprotectedHeaders));
            Assert.Null(ex.ParamName);
```